### PR TITLE
Registration confirm

### PR DESCRIPTION
**Pull request by micbaumr**
# Description

- Add toast configuration 
- Fix loader on start on web 
- make button to receive only icon names from icon set, apply styles to icons according on button variant and state
- add flexGap property to design system 
- add registration confirm 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Screenshot(s)

Please attach some screenshot or a movie if provided changes affects UI.

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [x] Add correct label to your pull request
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/design-system/components/Button/Button.tsx [hunk 1]
```diff
@@ -215,37 +227,64 @@ const RawButton = memo(
...
-                {children}
-              </Text>
-            )
-          }
-          return children
+          return (
+            <>
+              {loading ? (
+                <Loader type="default" size={24} />
+              ) : (
```
Comment (MateuszRostkowski) [inline src/design-system/components/Button/Button.tsx:234]: Previous conditional logic was more readable previously.

Additionally you don't need fragment here, it makes the code less readable.

Comment (MateuszRostkowski) [inline src/design-system/components/Button/Button.tsx:234]: Previous conditional logic was more readable.

Also you don't need fragment here.

File: src/screens/ConfirmEmail.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    <Center flexGrow={1}>
+      {isLoading && (
+        <Center gap={6}>
+          <Loader type="bubbles" />
+          <Text>Trwa weryfikowanie adresu e-mail ...</Text>
+        </Center>
+      )}
+      {isError && (
+        <Center>
+          <Text>Ups nie udało się zweryfikować adresu email.</Text>
```
Comment (MateuszRostkowski) [inline src/screens/ConfirmEmail.tsx:37]: Translations, you can add todo comment to add it later.

File: src/screens/ConfirmEmail.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      }
+    }
+  }, [code, confirmEmailMutation])
+
+  return (
+    <Center flexGrow={1}>
+      {isLoading && (
+        <Center gap={6}>
+          <Loader type="bubbles" />
+          <Text>Trwa weryfikowanie adresu e-mail ...</Text>
```
Comment (MateuszRostkowski) [inline src/screens/ConfirmEmail.tsx:32]: Translations, you can add todo comment to add it later.

File: src/screens/ConfirmEmail.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      {isError && (
+        <Center>
+          <Text>Ups nie udało się zweryfikować adresu email.</Text>
+          <Spacer y={4} />
+          <Button onPress={navigateToSignIn}>Wróć na stronę logowania</Button>
+        </Center>
+      )}
+      {isSuccess && (
+        <Center>
+          <Text>Adres e-mail został pomyślnie potwierdzony.</Text>
```
Comment (MateuszRostkowski) [inline src/screens/ConfirmEmail.tsx:44]: Translations, you can add todo comment to add it later.

File: src/screens/ConfirmEmail.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+          <Spacer y={4} />
+          <Button onPress={navigateToSignIn}>Wróć na stronę logowania</Button>
```
Comment (MateuszRostkowski) [inline src/screens/ConfirmEmail.tsx:46]: Translations, you can add todo comment to add it later.

File: src/screens/ConfirmEmail.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        <Center gap={6}>
+          <Loader type="bubbles" />
+          <Text>Trwa weryfikowanie adresu e-mail ...</Text>
+        </Center>
+      )}
+      {isError && (
+        <Center>
+          <Text>Ups nie udało się zweryfikować adresu email.</Text>
+          <Spacer y={4} />
+          <Button onPress={navigateToSignIn}>Wróć na stronę logowania</Button>
```
Comment (MateuszRostkowski) [inline src/screens/ConfirmEmail.tsx:39]: Translations, you can add todo comment to add it later.

**Review by MateuszRostkowski — APPROVED** [review]
In overall it looks good.

Please add translations or todo comments. 

Additionally please check my suggestions about `Button` component.

File: src/design-system/components/Button/Button.tsx (hunk 1 above)
Comment (micbaumr) [reply src/design-system/components/Button/Button.tsx]: Good one! I did it to look more like a previous logic 😄

**Pull request merged**